test/functional/mempool_reorg.py

```diff
@@ -8,36 +8,32 @@
 that spend (directly or indirectly) coinbase transactions.
 """
 
-from test_framework.blocktools import create_raw_transaction
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal, assert_raises_rpc_error
-
+from test_framework.wallet import MiniWallet
 
 class MempoolCoinbaseTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 2
+        self.setup_clean_chain = True
```

Comment:
Generating blocks takes a few seconds in valgrind, so I am thinking if this test may benefit from using `miniwallet.scan_blocks` (to be introduced in #21200)

Reply:
I had a go but I don't think it helps in this case since the pre-mined chain doesn't send the coinbase transaction to `self._address`. I made it little bit faster though by only mining 100 blocks instead of 200.

Reply:
It should. Specifically the last 25 mature blocks (including block 100). And the last 25 immature blocks (including block 200), but you don't need those

Reply:
Sorry, there was a bug. If you rebase, the bug should fix itself.

Reply:
yes, it worked! I updated the PR to use `miniwallet.scan_blocks` starting from block 76.